semgrep-core/src/core/il/IL_lvalue_helpers.ml

```diff
@@ -21,7 +21,14 @@ open IL
 
 let compare_name x y =
   let ident_cmp = String.compare (fst x.ident) (fst y.ident) in
-  if ident_cmp <> 0 then ident_cmp else Int.compare x.sid y.sid
+  if ident_cmp <> 0 then ident_cmp
+    (* If their names match, and either of the names have an unresolved sid,
+       then let's just say that they are the same.
+       In particular, this will help with Terraform, where we purposefully introduce
+       unresolved names as the parameters to fake functions.
```

Comment:
For DeepSemgrep, I think we should instead resolve the name that is not being resolved.

Reply:
As far as I can tell, DeepSemgrep only works with two types of resolved names, corresponding to `AST_generic.LocalVar` and `AST_generic.ResolvedName`. If the name you are interested in is already resolved, and it's local, you can just declare the parameter as the same local name (it seems that DeepSemgrep doesn't distinguish  between `Parameter` and `LocalVar` as Semgrep does). If that name is global, you can transform it into a local by updating the `(AST_generic.id_info_id, IdInfo.t) Hashtbl.t`.